api/v1alpha1/common_types.go

```diff
@@ -218,6 +218,51 @@ func InitializeLoadBalancerSpec(spec *LoadBalancerSpec, def defaultLoadBalancerS
 	return spec
 }
 
+// NLBLoadBalancerSpec configures the AWS NLB load balancer for the component
+type NLBLoadBalancerSpec struct {
+	// Enables/disbles use of proxy protocol in the load balancer
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	ProxyProtocol *bool `json:"proxyProtocol,omitempty"`
+	// Enables/disables cross zone load balancing
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	CrossZoneLoadBalancingEnabled *bool `json:"crossZoneLoadBalancingEnabled,omitempty"`
+	// The list of optional Elastic IPs allocations
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	EIPAllocations []string `json:"eipAllocations,omitempty"`
+}
+
+type defaultNLBLoadBalancerSpec struct {
+	CrossZoneLoadBalancingEnabled, ProxyProtocol *bool
+	EIPAllocations                               []string
+}
+
+// Default sets default values for any value not specifically set in the NLBLoadBalancerSpec struct
+func (spec *NLBLoadBalancerSpec) Default(def defaultNLBLoadBalancerSpec) {
+	spec.ProxyProtocol = boolOrDefault(spec.ProxyProtocol, def.ProxyProtocol)
+	spec.CrossZoneLoadBalancingEnabled = boolOrDefault(spec.CrossZoneLoadBalancingEnabled, def.CrossZoneLoadBalancingEnabled)
+}
```

Comment:
I think it would be good to initialize also `EIPAllocations` to an empty list `[]string{}`. If not you will have to check if the field is nil each time you try to access it.
```suggestion
func (spec *NLBLoadBalancerSpec) Default(def defaultNLBLoadBalancerSpec) {
	spec.ProxyProtocol = boolOrDefault(spec.ProxyProtocol, def.ProxyProtocol)
	spec.CrossZoneLoadBalancingEnabled = boolOrDefault(spec.CrossZoneLoadBalancingEnabled, def.CrossZoneLoadBalancingEnabled)
	spec.EIPAllocations = []string{}
}
```

Reply:
Forget this, I saw where it is used, it makes sense as it is.